### PR TITLE
CharSequence instead of String for request parsers #110

### PR DESCRIPTION
**Pull request by YamStranger**

### CODE REVIEW

**Comment by YamStranger** [comment]
@yegor256 #128 done

**Comment by YamStranger** [comment]
@yegor256 #128 done

**Comment by davvd** [comment]
Thanks, someone will review your pull request soon

**Comment by davvd** [comment]
@pinaf review this one, please

**Comment by pinaf** [comment]
@YamStranger there are checkstyle violations in the travis build.
>
[ERROR] /src/main/java/org/takes/rq/RqForm.java[142]: Empty line inside method (EmptyLinesCheck)
[ERROR] /src/main/java/org/takes/rq/RqFake.java[76]: Line is longer than 80 characters (found 85). (LineLengthCheck)
[ERROR] /src/main/java/org/takes/rq/RqCookies.java[65]: Line is longer than 80 characters (found 81). (LineLengthCheck)

File: src/main/java/org/takes/rq/RqFake.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public RqFake(final String method) {
      * @param method HTTP method
      * @param query HTTP query
      */
-    public RqFake(final String method, final String query) {
+    public RqFake(final CharSequence method, final String query) {
```
Comment (pinaf) [inline src/main/java/org/takes/rq/RqFake.java]: @YamStranger what about parameter `query`?

File: src/main/java/org/takes/rq/RqFake.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public RqFake(final String method, final String query) {
      * @param query HTTP query
      * @param body HTTP body
      */
-    public RqFake(final String method, final String query, final String body) {
+    public RqFake(final CharSequence method, final String query, final String body) {
```
Comment (pinaf) [inline src/main/java/org/takes/rq/RqFake.java]: @YamStranger what about parameter `query`?

File: src/main/java/org/takes/rq/RqForm.java [hunk 3]
```diff
@@ -137,10 +137,11 @@ public RqForm(final Request req) throws IOException {
      * @param txt Text
      * @return Decoded
      */
-    private static String decode(final String txt) {
+    private static String decode(final CharSequence txt) {
         try {
+
             return URLDecoder.decode(
-                txt, Charset.defaultCharset().name()
+                    txt.toString(), Charset.defaultCharset().name()
```
Comment (pinaf) [inline src/main/java/org/takes/rq/RqForm.java]: @YamStranger bad indentation here.

**Comment by pinaf** [comment]
@YamStranger It seems like some methods in `RqFake` still have `String` parameters.
@yegor256 please confirm all of them should be changed.

**Comment by pinaf** [comment]
@YamStranger 5 comments above.

**Comment by yegor256** [comment]
@pinaf @YamStranger I confirm, all of them should be changed

**Comment by YamStranger** [comment]
@yegor256 fixes provided according comments from pinaf, issue #110

**Comment by yegor256** [comment]
@YamStranger try to run `mvn clean install -Pqulice`, you will see some issues

**Comment by YamStranger** [comment]
@yegor256 Thanks

**Comment by YamStranger** [comment]
Please review

**Comment by YamStranger** [comment]
@yegor256 Will be .gitattributes added to project like here https://github.com/jcabi/jcabi-aether (https://github.com/jcabi/jcabi-aether/issues/57)?

**Comment by yegor256** [comment]
@YamStranger maybe in the future... do we need it already?

File: src/main/java/org/takes/rq/RqCookies.java [hunk 4]
```diff
@@ -60,9 +60,10 @@ public RqCookies(final Request req) {
      * @return List of values (can be empty)
      * @throws IOException If fails
      */
-    public Iterable<String> cookie(final String key) throws IOException {
+    public Iterable<String> cookie(final CharSequence key) throws IOException {
         final Map<String, String> map = this.map();
-        final String value = map.get(key.toLowerCase(Locale.ENGLISH));
+        final String keyAsString = key.toString().toLowerCase(Locale.ENGLISH);
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqCookies.java]: see http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

File: src/main/java/org/takes/rq/RqFake.java [hunk 5]
```diff
@@ -73,7 +73,8 @@ public RqFake(final String method, final String query) {
      * @param query HTTP query
      * @param body HTTP body
      */
-    public RqFake(final String method, final String query, final String body) {
+    public RqFake(final CharSequence method, final CharSequence query,
+                  final CharSequence body) {
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqFake.java]: why so many indentation spaces? should be four, as everywhere else

**Comment by yegor256** [comment]
@YamStranger a few comments above

File: src/main/java/org/takes/rq/RqFake.java (hunk 5 above)
Comment (YamStranger) [inline src/main/java/org/takes/rq/RqFake.java]: @yegor256 during last review, @pinaf asked me to fix checkstyle violations. 
[ERROR] /src/main/java/org/takes/rq/RqFake.java[76]: Line is longer than 80 characters (found 85). (LineLengthCheck)
I change style of method declaration - add "new line char" after "query,". I like when parameters of method starts on one vertical line, as for me it is improve readiness of code. 

As I understand you want me to do like like here  /org/takes/facets/fork/FkParams.java:132
I will fix it

Comment (yegor256) [inline src/main/java/org/takes/rq/RqFake.java]: @YamStranger look at other classes and you will see which code style we're using. we indent by 4 spaces always, no exceptions

**Comment by YamStranger** [comment]
@yegor256 I provided fix according your comments. About .gitattributes - yes, it makes life easier.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@pinaf According to our updated [quality rules](http://www.teamed.io/qa.html) performer of the ticket is responsible for all fixes in the ticket.

> Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

There is no description provided. Pull request must briefly explain what was done in the branch, so that we can use it in future. Could you please fix it?

**Comment by elenavolokhova** [comment]
>davvd commented 4 days ago
Thanks, someone will review your pull request soon

>YamStranger commented 3 days ago
Please review

@YamStranger @davvd Please don't forget to address the comments in the future.

**Comment by YamStranger** [comment]
@pinaf, @elenavolokhova I added description, please take a look

**Comment by elenavolokhova** [comment]
@YamStranger Thank you!

**Comment by pinaf** [comment]
@elenavolokhova I'm the reviewer. That message is for @YamStranger  I suppose :)

**Comment by elenavolokhova** [comment]
> @elenavolokhova I'm the reviewer. That message is for @YamStranger I suppose :)

@yegor256 Please help to explain this.

**Comment by yegor256** [comment]
@pinaf Elena is right here, you're the performer of this ticket, you're responsible for its quality, you get the payment what it's closed. So, when you start a review you should bring up all quality concerns and make sure the author of the pull request addresses them. Basically you're in charge here, that's why all quality blames are on you :)

**Comment by pinaf** [comment]
@yegor256 ok I wasn't aware that it is the reviewer's responsibility to check ticker quality also :) thanks.

**Comment by elenavolokhova** [comment]
@pinaf
>davvd commented 4 days ago
Thanks, someone will review your pull request soon

>YamStranger commented 3 days ago
Please review

 According to our [quality rules](http://www.teamed.io/qa.html):
 > Messages in a ticket always start with a name of a user they are addressed to.
 
 As far as you are the performer of this ticket you are responsible for its quality. Please make sure that everyone follows this rule in your tickets in the future.
 
 Please confirm that everything is clear and you have no questions regarding this approach.

**Comment by pinaf** [comment]
@yegor256 this was closed 5 days ago

**Comment by yegor256** [comment]
@pinaf we're not paying until QA gives a green light, see: http://www.teamed.io/qa.html

**Comment by pinaf** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@pinaf Thank you!

**Comment by elenavolokhova** [comment]
@davvd Ticket passed QA review. Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Ticket passed QA review. Quality is acceptable.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
@pinaf I added 10 mins to @elenavolokhova (for QA review) in transaction 55492051

Much appreciated! **22 mins** was added to your account, payment ID `AP-34S308087D440122A`, time spent is 39 hours and 35 mins

you're getting extra minutes here (c=7)

+22 added to your rating, at the moment it is: [+7026](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**